libavcodec/x86/vvc/vvc_sad.asm

```diff
@@ -189,5 +189,125 @@ cglobal vvc_sad_32_128, 6, 9, 13, src1, src2, dx, dy, block_w, block_h, off1, of
 
     RET
 
+
+cglobal vvc_sad, 6, 9, 13, src1, src2, dx, dy, block_w, block_h, off1, off2, row_idx
+    INIT_OFFSET src1q, src2q, dxq, dyq, off1q, off2q
+    pxor               m3, m3
+    pxor               m8, m8
+
+    cmp          block_wd, 16
+    je         vvc_sad_16
+    cmp          block_wd, 32
+    jge    vvc_sad_32_128
+
+    vvc_sad_8:
+        .loop_height:
+        movu              xm0, [src1q]
```

Comment:
how about we process two lines at a time? if needed handle the last line at the end.
you can reuse a function(macro) for width 8 and 16.  
%macro VVC_SAD
loop_height:
    LOAD
   SAD
   DEC
   jmp loop_height
%end macro